### PR TITLE
Implemented dynamic logic for updating the notification schedule.

### LINKED ISSUE #3765: [UBS] The schedule of notifications
Add the ability to change the notification schedule. Settings must be saved in the database.

### PR DESCRIPTION
**Pull request by ryhaldenys**
# GreenCityUBS PR
Implemented dynamic logic for updating the notification schedule. Now, an admin can update the notification schedule, and the notificator will be restarted to work according to the updated schedule.

## Summary Of Changes :fire:
- Implemented required methods for notifying in the NotificationService.
- Implemented  the new package with notificators which start schedule for according type.
- Implemented  planner class which with the start of program starts scheduled tasks and provide oportunity to restart notificator according to type.
- Impelemted tests for all new methods and classes.
- Improved endpoint which send notifications to user's email.

## Issue Link
[#3765](https://github.com/ita-social-projects/GreenCity/issues/3765)

# PR Checklist Forms
- [ ] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells or duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: dao/src/main/java/greencity/repository/NotificationTemplateRepository.java [hunk 1]
```diff
@@ -24,4 +24,14 @@ public interface NotificationTemplateRepository extends JpaRepository<Notificati
...
+
+    /**
+     * method, that finds schedule of active Notification by Notification type.
+     *
+     * @return {@link String}
+     * @author Denys Ryhal
+     */
+    @Query("SELECT t.schedule FROM NotificationTemplate t "
+        + "WHERE t.notificationType = ?1 AND t.notificationStatus = 'ACTIVE'")
+    String findScheduleOfActiveTemplateByType(NotificationType type);
```
Comment (softservedata) [inline dao/src/main/java/greencity/repository/NotificationTemplateRepository.java]: Spring data / JPA doesn't like  property names on the ?1 parameter.
https://stackoverflow.com/questions/10802798/spring-data-jpa-query-with-parameter-properties
https://spring.io/blog/2014/07/15/spel-support-in-spring-data-jpa-query-definitions

File: dao/src/main/java/greencity/repository/OrderRepository.java [hunk 2]
```diff
@@ -181,4 +182,41 @@ void updateOrderStatusToExpected(@Param("actual_status") String actualStatus,
...
+     */
+    @Query("select o from Order o "
+        + "inner join fetch o.events e WHERE e.eventName IN (:eventNames)")
+    List<Order> findAllWithEventsByEventNames(@Param("eventNames") String... eventNames);
+
+    /**
+     * method gets orders by order payment status left join events.
+     */
+    @Query("select o from Order o "
+        + "left join fetch o.events e WHERE o.orderPaymentStatus =?1")
```
Comment (softservedata) [inline dao/src/main/java/greencity/repository/OrderRepository.java]: Spring data / JPA doesn't like property names on the ?1 parameter.

File: service/src/test/java/greencity/ModelUtils.java [hunk 3]
```diff
@@ -200,6 +203,9 @@
 import static java.util.Collections.singletonList;
 
 public class ModelUtils {
+    private static final LocalDateTime LOCAL_DATE_TIME = LocalDateTime.of(1994, 3, 28, 15, 10);
```
Comment (softservedata) [inline service/src/test/java/greencity/ModelUtils.java]: What is this date for?

**Review by softservedata — APPROVED** [review]

File: service/src/test/java/greencity/ModelUtils.java (hunk 3 above)
Comment (ryhaldenys) [reply service/src/test/java/greencity/ModelUtils.java]: Fixed that.

File: dao/src/main/java/greencity/repository/OrderRepository.java (hunk 2 above)
Comment (ryhaldenys) [reply dao/src/main/java/greencity/repository/OrderRepository.java]: Fixed that.

File: dao/src/main/java/greencity/repository/NotificationTemplateRepository.java (hunk 1 above)
Comment (ryhaldenys) [reply dao/src/main/java/greencity/repository/NotificationTemplateRepository.java]: Fixed that.

**Review by KizerovDmitriy — APPROVED** [review]

**Review by urio999 — APPROVED** [review]

**Pull request merged**